cpp/cmake_modules/SetupCxxFlags.cmake

```diff
@@ -231,6 +231,26 @@ if (APPLE)
   set(CXX_COMMON_FLAGS "${CXX_COMMON_FLAGS} -stdlib=libc++")
 endif()
 
+if (CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64|AARCH64")
+  CHECK_CXX_SOURCE_COMPILES("
```

Comment:
Is this required?

By the way there are CRC intrinsics for ARM64 and an accompanying preprocessor macro. See example at https://stackoverflow.com/questions/37066261/why-is-arm-feature-crc32-not-being-defined-by-the-compiler

(more doc at https://gcc.gnu.org/onlinedocs/gcc/ARM-C-Language-Extensions-_0028ACLE_0029.html#ARM-C-Language-Extensions-_0028ACLE_0029 apparently)

Reply:
Thanks for your comments.
First I request crc extension capabilities from the assembler and detect the flags "-march=armv8-a+crc" for Arm64 crc intrinsics. I think it's also a reliable solution.  :)

Reply:
But the point is, it adds gratuitous complication. Testing for `-march=armv8-a+crc` should be sufficient.

Reply:
For not all compilers  support the Armv8 crc intrinsics, it is not sufficient just to test for 
`-march=armv8-a+crc `.  
If some compiler do not support intrinsics,  it is needed to request crc extension capabilities from the assembler.  So Arm asm : `crc32cx` will be tested first, and then we test for `-march=armv8-a+crc` to detect whether compiler supports intrinsics or not.
```
1. not support Arm crc
2. support Arm crc, but not support intrinsics
3. support intrinsics
```

Reply:
Which compilers don't support intrinsics? Apparently they are a ARM standard.